src/xbot/common/command/scripted/ScriptedCommand.java

```diff
@@ -0,0 +1,222 @@
+        }
+    }
+    
+    @AssistedInject
+    public ScriptedCommand(
+            XScheduler scheduler,
+            @Assisted("manualScriptText") String manualScriptText,
+            @Assisted("manualScriptName") String manualScriptName,
+            @Assisted ScriptedCommandFactory availableCommandFactory) {
+        
+        this.scheduler = scheduler;
+        this.availableCommandFactory = availableCommandFactory;
+        
+        this.manualScriptText = manualScriptText;
+        this.manualScriptName = manualScriptName;
+        
+        if(manualScriptText == null || manualScriptName == null) {
+            log.warn("Instantiated with null script or script name!");
+        }
+        else {
+            log.info("Instantiated with script string (" + manualScriptName + " is " + manualScriptText.length() + " chars)");
+        }
+    }
+    
+    private void initializeExecThread() {
+        log.info("Initializing exec thread");
+        
+        if(manualScriptText != null && manualScriptName != null) {
+            log.info("Using manual script text for exec");
+            execThread = new ScriptedCommandThread(manualScriptText, manualScriptName, this, availableCommandFactory);
+        }
+        else if (scriptFile != null && scriptFile.canRead() && scriptFile.isFile()) {
+            log.info("Using file path for exec");
+            execThread = new ScriptedCommandThread(scriptFile, this, availableCommandFactory);
+        }
+        else {
+            log.error("Exec thread not initialized due to lack of viable script source.");
+            return;
+        }
+        
+        execThread.setUncaughtExceptionHandler(new UncaughtExceptionHandler() {
+            
+            @Override
+            public void uncaughtException(Thread t, Throwable e) {
+                Logger log = Logger.getLogger(ScriptedCommandThread.class);
+                log.error("Uncaught exception in autonomous thread! " + e.toString());
+                e.printStackTrace();
+            }
+        });
+    }
+    
+    /**
+     * For internal use only!
+     * @param command The command to run
+     */
+    public synchronized void invokeCommand(BaseCommand command) {
+        log.info("Adding command " + command + " to scheduler");
+        
+        CancellableCommandWrapper wrappedCommand = new CancellableCommandWrapper(command);
+        
+        this.scheduler.add(wrappedCommand);
+        if(this.invokedCommands == null) {
+            this.invokedCommands = new HashSet<BaseCommand>();
+        }
+        
+        this.invokedCommands.add(wrappedCommand);
+    }
+    
+    @Override
+    public void initialize() {
+        log.info("Initialized");
+        initializeExecThread();
+    }
+
+    @Override
+    public void execute() {
+        // Start the execution thread if this is the first command run
+        if(execThread != null && execThread.getState() == State.NEW) {
```

Comment:
@aschokking because it starts the script immediately, so I don't want to start it until exec is called.